react-intl/index.d.ts

```diff
@@ -12,7 +12,7 @@ declare namespace ReactIntl {
         pluralRuleFunction?: (n: number, ord: boolean) => string;
     }
 
-    function injectIntl<TOriginalProps, TOwnProps>(component: React.ComponentClass<TOriginalProps> | React.StatelessComponent<TOriginalProps>): React.ComponentClass<TOwnProps>;
+    function injectIntl<T>(component: React.ComponentClass<T> | React.StatelessComponent<T>): React.ComponentClass<T>;
```

Comment:
the injectIntl declaration isnt working as class decorator

`@injectIntl`
`export class MyComponent extends React.Component<MyComponentProps, MyComponentState> {`
`...`
`}`

following would fix that problem:

`function injectIntl<T extends React.ComponentClass<InjectedIntlProps> | React.StatelessComponent<InjectedIntlProps>>(component: T) : T;`

or 

`function injectIntl<T extends React.ComponentClass<{}> | React.StatelessComponent<{}>>(component: T) : T;`